components/fxa-client/src/internal/state_persistence.rs

```diff
@@ -190,15 +201,15 @@ impl From<StateV1> for Result<StateV2> {
     }
 }
 
-// `StateV1` was a previous state schema.
-//
-// The below is sufficient to read existing state data serialized in this form, but should not
-// be used to create new data using that schema, so it is deliberately private and deliberately
-// does not derive(Serialize).
-//
-// If you find yourself modifying this code, you're almost certainly creating a potential data-migration
-// problem and should reconsider.
-
+/// `StateV1` was a previous state schema.
```

Comment:
Does this need to exist still? It looks like it's very old and we will never hit this case in the wild. (OTOH, no need to block - maybe another ticket which could be good for onboarding?)

Reply:
Sounds good, filed as https://github.com/mozilla/application-services/issues/3912